File: plugins/mixology/src/main/scala/com/fredplugins/mixology/FredsMixologyOverlay.java
package com.fredplugins.mixology;

import net.runelite.client.ui.overlay.Overlay;
import net.runelite.client.ui.overlay.OverlayLayer;
import net.runelite.client.ui.overlay.OverlayPosition;
import net.runelite.client.ui.overlay.outline.ModelOutlineRenderer;

import javax.inject.Inject;
import java.awt.Dimension;
import java.awt.Graphics2D;

public class FredsMixologyOverlay extends Overlay {

    private final FredsMixologyPlugin plugin;
    private final ModelOutlineRenderer modelOutlineRenderer;

    @Inject
    FredsMixologyOverlay(FredsMixologyPlugin plugin, ModelOutlineRenderer modelOutlineRenderer) {
        this.plugin = plugin;
        this.modelOutlineRenderer = modelOutlineRenderer;
        setPosition(OverlayPosition.DYNAMIC);
        setLayer(OverlayLayer.ABOVE_SCENE);
    }

    @Override
    public Dimension render(Graphics2D graphics) {
        for (var highlightedObject : plugin.highlightedObjects().values()) {
            modelOutlineRenderer.drawOutline(highlightedObject.object(), highlightedObject.outlineWidth(), highlightedObject.color(), highlightedObject.feather());
        }
        return null;
    }
}

File: plugins/mixology/src/main/scala/com/fredplugins/mixology/FredsMixologyOverlay.scala
package com.fredplugins.mixology

import com.google.inject.Inject
import net.runelite.client.ui.overlay.Overlay
import net.runelite.client.ui.overlay.OverlayLayer
import net.runelite.client.ui.overlay.OverlayPosition
import net.runelite.client.ui.overlay.outline.ModelOutlineRenderer

import java.awt.Dimension
import java.awt.Graphics2D
import scala.jdk.CollectionConverters.*
import scala.jdk.OptionConverters.*
import scala.jdk.StreamConverters.*
import scala.util.chaining.*
import scala.util.{Random, Try}
import scala.compiletime.uninitialized

class FredsMixologyOverlay @Inject()(val plugin: FredsMixologyPlugin, val modelOutlineRenderer: ModelOutlineRenderer) extends Overlay(plugin) {
	setPosition(OverlayPosition.DYNAMIC)
	setLayer(OverlayLayer.ABOVE_SCENE)

	override def render(graphics: Graphics2D): Dimension = {
		for ((alchobjEnum, highlightedObject) <- plugin.getHighlightedObjects) {
			modelOutlineRenderer.drawOutline(highlightedObject.`object`, highlightedObject.outlineWidth, highlightedObject.color, highlightedObject.feather)
		}

		null
	}
}
